### PR TITLE
Features/cas 12409 add css for cards zone

### PR DESCRIPTION
**Pull request by InesSimonica**
Created a functional recently played zone and added css to it

### CODE REVIEW

File: src/Components/card/Card.js [hunk 1]
```diff
@@ -2,12 +2,15 @@ import '../Style/Card.css'
...
 
 
-export const Card = ({ id, title, imageUrl, description}) => {
+export const Card = ({ id, title, imageUrl, description, AddRecentCard}) => {
     
     return(
-        <div className='card' onClick={() => OpenNewWindow(description)}>
+        <div className='card' onClick={() => {
+            OpenNewWindow(description)
+            AddRecentCard(id)
```
Comment (botondfd) [inline src/Components/card/Card.js]: please use camelCase while writing JS 😄

File: src/Components/homepage/Homepage.js [hunk 2]
```diff
@@ -2,34 +2,72 @@ import { useEffect, useState } from 'react'
...
     }, [])
 
+    // useEffect(() => {
+    //     if (cards.length) {
+    //         console.log(cards)
+    //         const firstCards = cards.slice(0,4)
+    //         setRecentlyPlayedCards(firstCards)
+    //     }
+    // }, [cards])
+
```
Comment (botondfd) [inline src/Components/homepage/Homepage.js]: can be removed if not used anymore

File: src/Components/homepage/Homepage.js [hunk 3]
```diff
@@ -2,34 +2,72 @@ import { useEffect, useState } from 'react'
...
+    const addRecentCard = (clickedCardId) => {
+        setRecentlyPlayedCards((oldCardsArray) => {
+            const found = oldCardsArray.find((element) => element === cards[clickedCardId - 1])
```
Comment (botondfd) [inline src/Components/homepage/Homepage.js]: `element === cards[clickedCardId - 1]` will this work if the cards are not ordered by `id`? 🤔 
a better solution would be:
```
const found = oldCardsArray.find((el) => el.id === clickedCardId);
```
important to know: while using array methods like `map`, `filter`, `find`, `some` you have access to the currently processed element as a whole, it is way more safe than accessing the elments by `[id]`

File: src/Components/homepage/Homepage.js [hunk 4]
```diff
@@ -2,34 +2,72 @@ import { useEffect, useState } from 'react'
...
+            if (found) {
+                return [...oldCardsArray]
+            }
```
Comment (botondfd) [inline src/Components/homepage/Homepage.js]: why do we return the old elements if the clicked one is found? 🤔 
i think you can just invert the condition and omit this step

File: src/Components/homepage/Homepage.js [hunk 5]
```diff
@@ -2,34 +2,72 @@ import { useEffect, useState } from 'react'
...
+            else {
+                if (oldCardsArray.length >= 4) {
```
Comment (botondfd) [inline src/Components/homepage/Homepage.js]: why is this a special case if there are more than 4 elements?

File: src/Components/homepage/Homepage.js [hunk 6]
```diff
@@ -2,34 +2,72 @@ import { useEffect, useState } from 'react'
...
+                    oldCardsArray.unshift(cards[clickedCardId - 1])
+                    oldCardsArray.pop()
```
Comment (botondfd) [inline src/Components/homepage/Homepage.js]: please try avoiding mutating the array directly, in general it is a good practice to treat arrays as immutable and create new ones if data needs to be changed

you could try refactoring this with "non-mutating" methods, like `.filter`

more on this topic: https://www.sitepoint.com/immutable-array-methods-javascript/

File: src/Components/homepage/Homepage.js [hunk 7]
```diff
@@ -2,34 +2,72 @@ import { useEffect, useState } from 'react'
...
+                    return [...oldCardsArray]
+                }
+                else {  
+                    oldCardsArray.unshift(cards[clickedCardId - 1])
+                    return [...oldCardsArray]
```
Comment (botondfd) [inline src/Components/homepage/Homepage.js]: same as above

File: src/Components/homepage/Homepage.js [hunk 8]
```diff
@@ -2,34 +2,72 @@ import { useEffect, useState } from 'react'
...
+        <div className='homepage-container'>
+            <div className= 'cards-container'>
+                {cards.map((element) =>
+                    <Card
+                        key={element.id}
+                        id={element.id}
+                        title={element.title}
+                        imageUrl={element.url}
+                        description={element.description}
+                        AddRecentCard={addRecentCard}
```
Comment (botondfd) [inline src/Components/homepage/Homepage.js]: `addRecentCard={addRecentCard}`
in general prop names should be camel cased

**Review by botondfd — CHANGES_REQUESTED** [review]

File: src/Components/Style/Recently-played-zone.css [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+
+.recently-played-cards-container-0 > .card,
+.recently-played-cards-container-1 > .card,
+.recently-played-cards-container-2 > .card,
+.recently-played-cards-container-3 > .card,
+.recently-played-cards-container-4 > .card {
+  height: 206px;
+  margin: 10px;
+  width: 170px;
+}
```
Comment (ionistratifd) [inline src/Components/Style/Recently-played-zone.css]: when having the same CSS rule for numbered containers or similar stuff, you can use a class wildcard like this
```suggestion
div[class^="recently-played-cards-container-"] > .card {
  height: 206px;
  margin: 10px;
  width: 170px;
}
```
this is checking if the class attribute starts with the string you provided, so the number at the end can be dynamic

Comment (ionistratifd) [reply src/Components/Style/Recently-played-zone.css]: the same comment applies to other instances on this CSS, you can find an explanation here: https://www.geeksforgeeks.org/wildcard-selectors-and-in-css-for-classes/

File: src/Components/Recently-played-zone/Recently-played-zone.js [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+                {receivedCards.map((element) =>
+                    <Card
+                        key={element.id}
+                        id={element.id}
+                        title={element.title}
+                        imageUrl={element.url}
+                        description={element.description}
+                        addRecentCard={addRecentCard}
+                    />
+                )}
```
Comment (ionistratifd) [inline src/Components/Recently-played-zone/Recently-played-zone.js]: you can use destructuring on cases like this, avoiding repeating every time the object name

```suggestion
                {receivedCards.map(({ id, title, url, description }) =>
                    <Card
                        key={id}
                        id={id}
                        title={title}
                        imageUrl={url}
                        description={description}
                        addRecentCard={addRecentCard}
                    />
                )}
```

File: src/Components/homepage/Homepage.js [hunk 11]
```diff
@@ -2,34 +2,72 @@ import { useEffect, useState } from 'react'
...
-                    id = {element.id}
-                    title = {element.title}
-                    imageUrl = {element.url}
-                    description = {element.description}
-                />
-            )}
-            
+        <div className='homepage-container'>
+            <div className= 'cards-container'>
+                {cards.map((element) =>
```
Comment (ionistratifd) [inline src/Components/homepage/Homepage.js]: same here with destructuring

File: src/Components/homepage/Homepage.js [hunk 12]
```diff
@@ -2,34 +2,72 @@ import { useEffect, useState } from 'react'
...
 	const numberOfCards = 20
 
     useEffect(() => {
-        fetchData(numberOfCards).then((res) => {setCards(res)}).catch((e) => {
-            setErrorMessage(e)
+        fetchData(numberOfCards)
+        .then((res) => {setCards(res)})
+        .catch((error) => {
+            setErrorMessage(error)
             console.error('[ERROR IN HOMEPAGE COMPONENT FETCH DATA]: ' + errorMessage)
```
Comment (ionistratifd) [inline src/Components/homepage/Homepage.js]: i don't think this log will display the correct errorMessage, as the value of `errorMessage` is still taken from the same render, so it's not hydrated yet, this just to be checked

**Review by ionistratifd — CHANGES_REQUESTED** [review]

**Review by botondfd — APPROVED** [review]

**Pull request merged**